app/controllers/api_controller/arbitration_profiles.rb

```diff
@@ -0,0 +1,49 @@
+class ApiController
+  module ArbitrationProfiles
+    def create_resource_arbitration_profiles(_type, _id, data)
+      validate_profile_data(data)
+      attributes = data.dup
+      fetch_set_availability_zone(attributes)
+      fetch_set_provider(attributes)
+      arbitration_profile = collection_class(:arbitration_profiles).create(attributes)
+      if arbitration_profile.invalid?
+        raise BadRequestError, "Failed to add new arbitration profile -
+            #{arbitration_profile.errors.full_messages.join(', ')}"
+      end
+      arbitration_profile
+    end
+
+    def edit_resource_arbitration_profiles(type, id, data)
+      validate_profile_data(data)
+      fetch_set_provider(data)
+      fetch_set_availability_zone(data)
+      edit_resource(type, id, data)
+    end
+
+    private
+
+    def fetch_set_provider(data)
+      provider_data = data.delete('provider') || data.delete('ext_management_system')
+      if provider_data
+        provider = fetch_provider(provider_data)
+        data.merge!('ext_management_system' => provider) if provider
+      end
+    end
+
+    def fetch_set_availability_zone(data)
+      availability_zone_data = data.delete('availability_zone')
```

Comment:
Even though we `dup`'d the data earlier, we're still changing the thing that's passed into this method. Is there a ways we could avoid doing that too? These fetch/set methods (which I know are a pattern that's been established in the API) break the command/query separation principle - I think it could be better to have them just do the fetch and to do the setting in the main method so the data is not passed around and mutated along the way. WDYT?